aptos-move/aptos-aggregator/src/delayed_field_extension.rs

```diff
@@ -546,66 +215,11 @@ impl AggregatorData {
-        // need to clone here, so we can call self.read_snapshot below.
-        let snapshot_state = match self.aggregator_snapshots.entry(snapshot_id) {
-            Entry::Vacant(entry) => {
-                let value_from_storage = get_delayed_field_value_from_storage(
-                    &snapshot_id,
-                    resolver,
-                    DelayedFieldReadMode::Aggregated,
-                )?;
-                entry
-                    .insert(AggregatorSnapshot {
-                        id: snapshot_id,
-                        state: AggregatorSnapshotState::Reference {
-                            speculative_value: SnapshotValue::try_from(value_from_storage)?,
-                        },
-                    })
-                    .state
-                    .clone()
-            },
-            Entry::Occupied(entry) => entry.get().state.clone(),
-        };
-        match snapshot_state {
-            AggregatorSnapshotState::Create { value } => Ok(value),
-            AggregatorSnapshotState::Delta {
-                base_aggregator,
-                delta,
-            } => match self
-                .aggregators
-                .get_mut(&AggregatorVersionedID::V2(base_aggregator))
-            {
-                Some(aggregator) => {
-                    // We need to make sure speculative_start_value is in a state for read,
-                    // but we need the value at the start of the transaction
-                    let speculative_start_value = aggregator.read_aggregated_aggregator_value(
-                        resolver,
-                        ReadPosition::BeforeCurrentTxn,
-                    )?;
-                    Ok(SnapshotValue::Integer(expect_ok(
-                        BoundedMath::new(aggregator.max_value)
-                            .unsigned_add_delta(speculative_start_value, &delta),
-                    )?))
-                },
-                None => Err(PartialVMError::from(code_invariant_error(
-                    "AggregatorSnapshotState::Delta without base aggregator being set",
-                ))),
-            },
-            AggregatorSnapshotState::Derived {
-                base_snapshot,
-                formula,
-            } => {
-                let base = self.read_snapshot(base_snapshot, resolver)?;
-                match base {
-                    SnapshotValue::Integer(v) => Ok(SnapshotValue::String(formula.apply_to(v))),
-                    SnapshotValue::String(_) => Err(PartialVMError::from(code_invariant_error(
-                        "Tried calling concat on String SnapshotValue",
-                    ))),
-                }
-            },
-            AggregatorSnapshotState::Reference { speculative_value } => Ok(speculative_value),
-        }
+        Ok(SnapshotValue::try_from(self.read_value(
+            snapshot_id,
+            resolver,
+            ReadPosition::AfterCurrentTxn,
```

Comment:
Ia this right? Shouldn't this be BeforeCurrentTxn?

Reply:
if snapshot is created in this transaction, there is no value for it for BeforeCurrentTxn. 

since it is immutable - AfterCurrentTxn is equivalent to current, and is equivalent to state it was snapshotted at